lib/msal-browser/README.md

```diff
@@ -38,14 +72,129 @@ We plan to continue support for the implicit flow in the library.
 ## Usage
 The example below walks you through how to login a user and acquire a token to be used for Microsoft's Graph Api.
 
-#### Prerequisite
+#### Prerequisites
 
 Before using MSAL.js you will need to [register an application in Azure AD](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) to get a valid `clientId` for configuration, and to register the routes that your app will accept redirect traffic on.
 
 > TBD
 
+#### 1. Initializing the Public Client Application
+
+`PublicClientApplication` can be configured with a variety of different options, detailed in our [Wiki](), but the only required parameter is `auth.clientId`.
+
+```javascript
+import * as msal from "msal";
+
+const msalConfig = {
+    auth: {
+        clientId: 'your_client_id'
+    }
+};
+
+const msalInstance = new msal.PublicClientApplication(msalConfig);
+```
+
+#### 2. Choose interaction type - redirect or popup
+
+Choose which APIs you will use in your authentication flows:
+- loginRedirect and acquireTokenRedirect
+- loginPopup and acquireTokenPopup
+
+If you are using the redirect APIs, you will need to include the helper function below with a valid callback API. If you do not use this, your application will error out if any of the redirect APIs are used. This is not needed for any popup APIs.
+
+```javascript
+msalInstance.handleRedirectCallback((error, response) => {
+    // handle redirect response or error
+});
+```
+
+#### 3. Login the user
+
+Your app must login the user with either the `loginPopup` or the `loginRedirect` method to establish user context.
+
+When the login methods are called and the authentication of the user is completed by the Azure AD service, an [id token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens) is returned which is used to identify the user with some basic information.
+
+When you login a user, you can pass in scopes that the user can pre consent to on login, however this is not required. Please note that consenting to scopes on login, does not return an access_token for these scopes, but gives you the opportunity to obtain a token silently with these scopes passed in, with no further interaction from the user.
```

Comment:
"that the user can pre-consent to, however, this is not required"

Reply:
done.